src/controllers/UserController.ts

```diff
@@ -49,90 +69,232 @@ export default class UserController implements IController {
-            serviceDataPermissions ? user.removeNonRequestedData(serviceDataPermissions) : user.removeSensitiveInformation()
+          new ServiceResponse(serviceDataPermissions ? user.removeNonRequestedData(serviceDataPermissions) : user.removeSensitiveInformation(), 'Success'
           )
         );
     } catch (e) {
-      res.status(e.httpErrorCode).json(new ServiceResponse(null, e.message));
+      return res
+        .status(e.httpErrorCode)
+        .json(new ServiceResponse(null, e.message));
     }
   }
 
-  async getAllUsers(req: any, res: express.Response) {
-    if (compareRoles(req.authorization.user.role, 'kayttaja') <= 0) {
+  /**
+   * Returns all users.
+   *
+   * @param {(express.Request | any)} req
+   * @param {express.Response} res
+   * @returns
+   * @memberof UserController
+   */
+  async getAllUsers(req: express.Request & IASRequest, res: express.Response) {
+    if (compareRoles(req.authorization.user.role, "kayttaja") <= 0) {
       return res.status(403).json(new ServiceResponse(null, "Forbidden"));
     }
 
     // Request is a search
     if (req.query.searchTerm) {
       try {
-        let users = await this.userService.searchUsers(req.query.searchTerm);
+        let users: User[] = await this.userService.searchUsers(
+          req.query.searchTerm
+        );
+        return res
+          .status(200)
+          .json(
+            new ServiceResponse(users.map(u => u.removeSensitiveInformation()))
+          );
+      } catch (e) {
+        return res.status(500).json(new ServiceResponse(null, e.message));
+      }
+    }
+
+    // Request is only looking for certain fields
+    if (req.query.fields) {
+      try {
+        let users: User[] = await this.userService.fetchAllWithSelectedFields(req.query.fields, req.query.conditions || null);
         return res
           .status(200)
           .json(
             new ServiceResponse(users.map(u => u.removeSensitiveInformation()))
           );
       } catch (e) {
-        res.status(500).json(new ServiceResponse(null, e.message));
+        return res.status(500).json(new ServiceResponse(null, e.message));
       }
-      return;
     }
 
     try {
-      let users = await this.userService.fetchAllUsers();
+      let users: User[] = await this.userService.fetchAllUsers();
       return res
         .status(200)
         .json(
           new ServiceResponse(users.map(u => u.removeSensitiveInformation()))
         );
     } catch (e) {
-      res.status(500).json(new ServiceResponse(null, e.message));
+      return res.status(500).json(new ServiceResponse(null, e.message));
     }
   }
 
-  async modifyMe(req: any, res: express.Response) {
-    if (req.params.id === 'me') {
+  /**
+   * Returns all unpaid users.
+   *
+   * @param {*} req
+   * @param {express.Response} res
+   * @returns
+   * @memberof UserController
+   */
+  async getAllUnpaidUsers(req: any, res: express.Response) {
+    if (req.authorization.user.role != "yllapitaja") {
```

Comment:
Use User::compareRoles to check permission. Also. I think even `jasevirkailija` has the rights to get unpaid users according to the old back-end.